lib/mapping/CMap.cpp

```diff
@@ -343,9 +347,15 @@ bool CMap::checkForVisitableDir(const int3 & src, const TerrainTile *pom, const
 
 		const CGObjectInstance * obj = pom->visitableObjects[b];
 
-		if (!obj->appearance.isVisitableFrom(src.x - dst.x, src.y - dst.y))
+		if(!obj->appearance.isVisitableFrom(src.x - dst.x, src.y - dst.y))
 			return false;
 	}
+
+	// TODO: This hacky code need improvement. It's used to not let hero exit from teleporter to wrong tile. E.g above teleporter.

```

Comment:
Oh and yeah it's not just affect teleporters actually, but I testing them and only them for that long time so didn't notice that.